src/components/Table.tsx

```diff
@@ -301,14 +301,20 @@ export default class Table<T> extends React.Component<
 		const checkedItems = filter(this.props.data, x =>
 			includes(selectedRowsIds, x[rowKey]),
 		);
-		const allChecked = checkedItems.length === this.props.data.length;
+		const allChecked = checkedItems.length === data.length;
 
 		this.setState({ allChecked, checkedItems });
 	};
 
 	toggleAllChecked = () => {
+		const { data } = this.props;
+
+		if (!data) {
+			return;
+		}
+
 		const allChecked = !this.state.allChecked;
-		const checkedItems = allChecked ? this.props.data.slice() : [];
+		const checkedItems = allChecked ? data.slice() : [];
```

Comment:
How about doing a `this.props.data || [])?
We might also want to have `allChecked = allChecked && !_.isEmpty(this.props.data)` to avoid an awkward situation that the all check marked is ticked but there are no data at all.

Reply:
Good idea, as above, I'll add a test case